File: src/main/java/no/capraconsulting/siren/internal/util/MapUtil.java
package no.capraconsulting.siren.internal.util;

import java.util.LinkedHashMap;
import java.util.Map;

public final class MapUtil {
    private MapUtil() {}

    public static <K, V> Map<K, V> skipNulls(Map<K, V> map) {
        Map<K, V> result = new LinkedHashMap<>();
        map.forEach((k, v) -> {
            if (v != null) {
                result.put(k, v);
            }
        });
        return result;
    }

    public static boolean notNull(Map<String, Object> map, String key) {
        return map.containsKey(key) && map.get(key) != null;
    }
}

File: src/main/kotlin/no/capraconsulting/siren/internal/util/MapUtil.kt
@file:JvmName("MapUtil")
package no.capraconsulting.siren.internal.util

@Deprecated("Use extension method",
    ReplaceWith("map.skipNulls()", "no.capraconsulting.siren.internal.util.MapUtil.skipNulls")
)
internal fun <K, V> skipNulls(map: Map<K, V>): Map<K, V> =
    map.skipNulls()

@JvmName("skipNullsFuture")
internal fun <K, V> Map<K, V>.skipNulls() =
    filterNot { it.value == null }

internal fun notNull(map: Map<String, Any>, key: String): Boolean {
    return map.containsKey(key) && map[key] != null
}
